src/libsyntax/codemap.rs

```diff
@@ -646,6 +646,63 @@ impl CodeMap {
         }
     }
 
+    /// Reformat `sp`'s snippet to oneline if it is available
+    ///
+    /// Given a snippet like:
+    ///
+    /// ```text
+    /// fn foo< 'lifetime, T >(
+    ///     &self,
+    ///     bar : &Type< 'lifetime, T>)
+    ///     -> std::result::Result<(),
+    ///         Error>;
+    /// ```
+    ///
+    /// it'll return:
+    ///
+    /// ```text
+    /// fn foo<'lifetime, T>(&self, bar: &Type<'lifetime, T>) -> std::result::Result<(), Error>;
+    /// ```
+    pub fn span_to_oneline_snippet(&self, sp: Span) -> Result<String, SpanSnippetError> {
```

Comment:
@nikomatsakis could there be a method already in the codebase that does this? If there isn't, does this seem reasonable to be added for these use cases?